Packs/AWS_SystemManager/Integrations/AWSSystemManager/AWSSystemManager.py

```diff
@@ -0,0 +1,1612 @@
+
+    Example:
+    -------
+        >>> args = {
+        >>>    'instance_id': 'i-0a00aaa000000000a', # valid instance id
+        >>>    'association_id': '0000' # invalid association id
+        >>> }
+        >>> try:
+        >>>    validate_args(args)
+        >>> except DemistoException as e:  # e equals to "Invalid association id: 0000"
+        >>>     print(f"Validation error: {e}")
+            Validation error: Invalid association id: 0000
+    """
+    for validator in (
+        ASSOCIATION_ID_VALIDATOR,
+        ASSOCIATION_VERSION_VALIDATOR,
+        INSTANCE_ID_VALIDATOR,
+    ):
+        if (arg_value := args.get(validator.name)) and not re.search(
+            validator.regex,
+            arg_value,
+        ):
+            raise DemistoException(
+                validator.error_template.format(**{validator.name: arg_value}),
+            )
+    return None
+
+
+def config_aws_session(args: dict[str, str], aws_client: AWSClient) -> "SSMClient":
+    """Configures an AWS session for the Lambda service,
+    Used in all the commands.
+
+    Args:
+    ----
+        args: A dictionary containing the configuration parameters for the session.
+                     - 'region' (str): The AWS region.
+                     - 'roleArn' (str): The ARN of the IAM role.
+                     - 'roleSessionName' (str): The name of the role session.
+                     - 'roleSessionDuration' (str): The duration of the role session.
+
+        aws_client: The AWS client used to configure the session.
+
+    Returns:
+    -------
+        AWS session (ssm client): The configured AWS session.
+    """
+    return aws_client.aws_session(
+        service=SERVICE_NAME,
+        region=args.get("region"),
+        role_arn=args.get("roleArn"),
+        role_session_name=args.get("roleSessionName"),
+        role_session_duration=args.get("roleSessionDuration"),
+    )
+
+
+def convert_datetime_to_iso(response) -> dict[str, Any]:
+    """Converts datetime objects in a response dictionary to ISO formatted strings.
+
+    Args:
+    ----
+        response (dict): The response dictionary.
+
+    Returns:
+    -------
+        dict: The response dictionary with datetime objects converted to ISO formatted strings.
+
+    Example:
+    -------
+        >>> response = {
+        >>>    'timestamp': datetime(2023, 8, 20, 12, 30, 0),
+        >>>    'data': {
+        >>>        'created_at': datetime(2023, 8, 19, 15, 45, 0)
+        >>>    }
+        >>> }
+        >>> iso_response = convert_datetime_to_iso(response)
+        >>> print(iso_response)
+            {
+            'timestamp': '2023-08-20T12:30:00',
+            'data': {
+                 'created_at': '2023-08-19T15:45:00'
+               }
+            }
+    """
+
+    def _datetime_to_string(obj: Any) -> str:
+        if isinstance(obj, datetime):
+            return obj.isoformat()
+        return str(obj)
```

Comment:
`raise TypeError` instead, and unit test

Reply:
Done